### ISSUE TITLE
Reading directories only

### ISSUE THREAD

**Issue by Zyles** [thephpleague/flysystem#431]
I need to access local but I only have read access to the root directory. When I create a new local adapter as per documentation it complains that it is not writable.

Am I doing something wrong or is the following or similar not possible:

```
try {
			/* Root dir is /home/httpd/myapp (not writable) */
			$filesystem = new Filesystem(new Adapter($this->app->config('base_path')));
			
			/* Subdirectories (all are writable) */
			$directories = [
				'/archive',
				'/import',
				'/import/ftp-upload',
				'/app/config'
			];

			foreach ($directories as $directory) {
				if (!$filesystem->has($directory)) {
					throw new Exception('Directory ' . $directory . ' is not writable.');		
				}
			}
			

		} catch(Exception $e) {
			echo "Error: " . $e->getMessage();
			die();
		}
```

This returns error from line 3:

```
The root path /home/httpd/myapp is not writable.
```

I kinda want to mount the top level directory and then check stuff beneath it instead of having to use a new adapter for every subdirectory or give write access to the root directory.

**Comment by frankdejonge** [comment]
@Zyles the local adapter needs write access to it's root to function. What you could to have the same effect, is use the Mount Manager to handle multiple instances which are per sub directory instances. Have you looked into that? http://flysystem.thephpleague.com/mount-manager/

**Comment by Zyles** [comment]
I will look more closely at the mount manager. But I still think it would be beneficial to have read only support like you do in a normal filesystem.

**Comment by SteamboatColorado** [comment]
OK, I think I have the same problem.  I also used to make root defined as "/" and pass in absolute paths.  As soon as I upgraded from 0.5.12 to 1.0 this was no longer available.  Was this change due to security issues?  

I think it is sometimes useful to use absolute paths verses relative to (flysystem) root would be very useful as an option.  And that is what I did in the previous version was use absolute paths quite a bit.  You guys do great work as you already know.
